### ISSUE TITLE
Endless upgrade/downgrade cycle for graphwiz

### ISSUE THREAD

**Issue by diegobersanetti** [conda/conda#12844]
### Checklist

- [X] I added a descriptive title
- [X] I searched open reports and couldn't find a duplicate

### What happened?

**System**: Ubuntu 20.04.6, conda 23.5.0, python 3.10.11

Doing a "conda update --all" brings to a cycle where the first time graphwiz gets updated (and new packages pulled), then the following time it is downgraded to the previous version, and the new packages uninstalled. Then the cycle repeats itself over.

**Upgrade log**:

Retrieving notices: ...working... done
Collecting package metadata (current_repodata.json): done
Solving environment: done

Package Plan

  environment location: /home/blind/Software/anaconda3


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    boost-cpp-1.73.0           |      h7f8727e_12          16 KB
    c-ares-1.19.0              |       h5eee18b_0         118 KB
    graphviz-2.50.0            |       h1b29801_1         2.6 MB
    libboost-1.73.0            |      h28710b8_12        13.8 MB
    libcurl-8.1.1              |       h91b91d3_0         394 KB
    libev-4.33                 |       h7f8727e_1         111 KB
    libiconv-1.16              |       h7f8727e_2         736 KB
    libnghttp2-1.52.0          |       ha637b67_1         671 KB
    libssh2-1.10.0             |       h37d81fd_2         292 KB
    openjpeg-2.4.0             |       h3ad879b_0         331 KB
    poppler-22.12.0            |       h381b16e_0        17.3 MB
    poppler-data-0.4.11        |       h06a4308_1         2.6 MB
    ------------------------------------------------------------
                                           Total:        38.8 MB

The following NEW packages will be INSTALLED:

  boost-cpp          pkgs/main/linux-64::boost-cpp-1.73.0-h7f8727e_12 
  c-ares             pkgs/main/linux-64::c-ares-1.19.0-h5eee18b_0 
  libboost           pkgs/main/linux-64::libboost-1.73.0-h28710b8_12 
  libcurl            pkgs/main/linux-64::libcurl-8.1.1-h91b91d3_0 
  libev              pkgs/main/linux-64::libev-4.33-h7f8727e_1 
  libiconv           pkgs/main/linux-64::libiconv-1.16-h7f8727e_2 
  libnghttp2         pkgs/main/linux-64::libnghttp2-1.52.0-ha637b67_1 
  libssh2            pkgs/main/linux-64::libssh2-1.10.0-h37d81fd_2 
  openjpeg           pkgs/main/linux-64::openjpeg-2.4.0-h3ad879b_0 
  poppler            pkgs/main/linux-64::poppler-22.12.0-h381b16e_0 
  poppler-data       pkgs/main/linux-64::poppler-data-0.4.11-h06a4308_1 

The following packages will be UPDATED:

  graphviz                                2.50.0-h3cd0ef9_0 --> 2.50.0-h1b29801_1 


**Downgrade log**:

Collecting package metadata (current_repodata.json): done                                                                                                                                                   
Solving environment: done                                                                                                                                                                                    
                                                                                                                                                                                                             
Package Plan                                                                                                                                                                                     
                                                                                                                                                                                                             
  environment location: /home/blind/Software/anaconda3                                                                                                                                                       
                                                                                                                                                                                                             

The following packages will be REMOVED:

  boost-cpp-1.73.0-h7f8727e_12
  c-ares-1.19.0-h5eee18b_0
  libboost-1.73.0-h28710b8_12
  libcurl-8.1.1-h91b91d3_0
  libev-4.33-h7f8727e_1
  libiconv-1.16-h7f8727e_2
  libnghttp2-1.52.0-ha637b67_1
  libssh2-1.10.0-h37d81fd_2
  openjpeg-2.4.0-h3ad879b_0
  poppler-22.12.0-h381b16e_0
  poppler-data-0.4.11-h06a4308_1

The following packages will be DOWNGRADED:

  graphviz                                2.50.0-h1b29801_1 --> 2.50.0-h3cd0ef9_0 



### Conda Info

```shell
active environment : base
    active env location : /home/blind/Software/anaconda3
            shell level : 1
       user config file : /home/blind/.condarc
 populated config files : /home/blind/.condarc
          conda version : 23.5.0
    conda-build version : 3.25.0
         python version : 3.10.11.final.0
       virtual packages : __archspec=1=x86_64
                          __glibc=2.31=0
                          __linux=5.15.0=0
                          __unix=0=0
       base environment : /home/blind/Software/anaconda3  (writable)
      conda av data dir : /home/blind/Software/anaconda3/etc/conda
  conda av metadata url : None
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /home/blind/Software/anaconda3/pkgs
                          /home/blind/.conda/pkgs
       envs directories : /home/blind/Software/anaconda3/envs
                          /home/blind/.conda/envs
               platform : linux-64
             user-agent : conda/23.5.0 requests/2.29.0 CPython/3.10.11 Linux/5.15.0-76-generic ubuntu/20.04.6 glibc/2.31
                UID:GID : 1000:1000
             netrc file : /home/blind/.netrc
           offline mode : False
```


### Conda Config

```shell
==> /home/blind/.condarc <==
channel_priority: flexible
channels:
  - defaults
```


### Conda list

```shell
# packages in environment at /home/blind/Software/anaconda3:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main    defaults
_openmp_mutex             5.1                       1_gnu    defaults
anaconda-client           1.11.3          py310h06a4308_0    defaults
anaconda-navigator        2.4.2           py310h06a4308_0    defaults
anaconda-project          0.11.1          py310h06a4308_0    defaults
atk-1.0                   2.36.0               ha1a6a79_0    defaults
attrs                     22.1.0          py310h06a4308_0    defaults
backports                 1.1                pyhd3eb1b0_0    defaults
backports.functools_lru_cache 1.6.4              pyhd3eb1b0_0    defaults
backports.tempfile        1.0                pyhd3eb1b0_1    defaults
backports.weakref         1.0.post1                  py_1    defaults
beautifulsoup4            4.12.2          py310h06a4308_0    defaults
boltons                   23.0.0          py310h06a4308_0    defaults
brotlipy                  0.7.0           py310h7f8727e_1002    defaults
bzip2                     1.0.8                h7b6447c_0    defaults
ca-certificates           2023.05.30           h06a4308_0    defaults
cairo                     1.16.0               hb05425b_4    defaults
certifi                   2023.5.7        py310h06a4308_0    defaults
cffi                      1.15.1          py310h5eee18b_3    defaults
chardet                   4.0.0           py310h06a4308_1003    defaults
charset-normalizer        2.0.4              pyhd3eb1b0_0    defaults
click                     8.0.4           py310h06a4308_0    defaults
clyent                    1.2.2           py310h06a4308_1    defaults
conda                     23.5.0          py310h06a4308_0    defaults
conda-build               3.25.0          py310h06a4308_0    defaults
conda-content-trust       0.1.3           py310h06a4308_0    defaults
conda-index               0.2.3           py310h06a4308_0    defaults
conda-pack                0.6.0              pyhd3eb1b0_0    defaults
conda-package-handling    2.1.0           py310h06a4308_0    defaults
conda-package-streaming   0.8.0           py310h06a4308_0    defaults
conda-repo-cli            1.0.41          py310h06a4308_0    defaults
conda-token               0.4.0              pyhd3eb1b0_0    defaults
conda-verify              3.4.2                      py_1    defaults
cryptography              39.0.1          py310h9ce1e76_0    defaults
dbus                      1.13.18              hb2f20db_0    defaults
defusedxml                0.7.1              pyhd3eb1b0_0    defaults
expat                     2.4.9                h6a678d5_0    defaults
filelock                  3.9.0           py310h06a4308_0    defaults
font-ttf-dejavu-sans-mono 2.37                 hd3eb1b0_0    defaults
font-ttf-inconsolata      2.001                hcb22688_0    defaults
font-ttf-source-code-pro  2.030                hd3eb1b0_0    defaults
font-ttf-ubuntu           0.83                 h8b1ccd4_0    defaults
fontconfig                2.14.1               h4c34cd2_2    defaults
fonts-anaconda            1                    h8fa9717_0    defaults
fonts-conda-ecosystem     1                    hd3eb1b0_0    defaults
freetype                  2.12.1               h4a9f257_0    defaults
fribidi                   1.0.10               h7b6447c_0    defaults
future                    0.18.3          py310h06a4308_0    defaults
gdk-pixbuf                2.42.10              h5eee18b_0    defaults
giflib                    5.2.1                h5eee18b_3    defaults
glib                      2.69.1               he621ea3_2    defaults
glob2                     0.7                pyhd3eb1b0_0    defaults
gobject-introspection     1.72.0          py310hbb6d50b_2    defaults
graphite2                 1.3.14               h295c915_1    defaults
graphviz                  2.50.0               h3cd0ef9_0    defaults
gst-plugins-base          1.14.1               h6a678d5_1    defaults
gstreamer                 1.14.1               h5eee18b_1    defaults
gtk2                      2.24.33              h73c1081_2    defaults
gts                       0.7.6                hb67d8dd_3    defaults
harfbuzz                  4.3.0                hf52aaf7_1    defaults
icu                       58.2                 he6710b0_3    defaults
idna                      3.4             py310h06a4308_0    defaults
jinja2                    3.1.2           py310h06a4308_0    defaults
jpeg                      9e                   h5eee18b_1    defaults
jsonpatch                 1.32               pyhd3eb1b0_0    defaults
jsonpointer               2.1                pyhd3eb1b0_0    defaults
jsonschema                4.17.3          py310h06a4308_0    defaults
jupyter_core              5.3.0           py310h06a4308_0    defaults
krb5                      1.19.4               h568e23c_0    defaults
lcms2                     2.12                 h3be6417_0    defaults
ld_impl_linux-64          2.38                 h1181459_1    defaults
lerc                      3.0                  h295c915_0    defaults
libarchive                3.6.2                hb4bd9a0_1    defaults
libclang                  14.0.6          default_hc6dbbc7_1    defaults
libclang13                14.0.6          default_he11475f_1    defaults
libdeflate                1.17                 h5eee18b_0    defaults
libedit                   3.1.20221030         h5eee18b_0    defaults
libevent                  2.1.12               h8f2d780_0    defaults
libffi                    3.4.4                h6a678d5_0    defaults
libgcc-ng                 11.2.0               h1234567_1    defaults
libgd                     2.3.3                h6a678d5_2    defaults
libgomp                   11.2.0               h1234567_1    defaults
liblief                   0.12.3               h6a678d5_0    defaults
libllvm14                 14.0.6               hdb19cb5_3    defaults
libpng                    1.6.39               h5eee18b_0    defaults
libpq                     12.9                 h16c4e8d_3    defaults
librsvg                   2.54.4               h36cc946_3    defaults
libstdcxx-ng              11.2.0               h1234567_1    defaults
libtiff                   4.5.0                h6a678d5_2    defaults
libtool                   2.4.6             h6a678d5_1009    defaults
libuuid                   1.41.5               h5eee18b_0    defaults
libwebp                   1.2.4                h11a3e52_1    defaults
libwebp-base              1.2.4                h5eee18b_1    defaults
libxcb                    1.15                 h7f8727e_0    defaults
libxkbcommon              1.0.1                h5eee18b_1    defaults
libxml2                   2.10.3               hcbfbd50_0    defaults
libxslt                   1.1.37               h2085143_0    defaults
lz4-c                     1.9.4                h6a678d5_0    defaults
markupsafe                2.1.1           py310h7f8727e_0    defaults
more-itertools            8.12.0             pyhd3eb1b0_0    defaults
navigator-updater         0.4.0           py310h06a4308_0    defaults
nbformat                  5.7.0           py310h06a4308_0    defaults
ncurses                   6.4                  h6a678d5_0    defaults
ninja                     1.10.2               h06a4308_5    defaults
ninja-base                1.10.2               hd09550d_5    defaults
nspr                      4.35                 h6a678d5_0    defaults
nss                       3.89.1               h6a678d5_0    defaults
openssl                   1.1.1u               h7f8727e_0    defaults
packaging                 23.0            py310h06a4308_0    defaults
pango                     1.50.7               h05da053_0    defaults
patch                     2.7.6             h7b6447c_1001    defaults
patchelf                  0.17.2               h6a678d5_0    defaults
pathlib                   1.0.1              pyhd3eb1b0_1    defaults
pcre                      8.45                 h295c915_0    defaults
pillow                    9.4.0           py310h6a678d5_0    defaults
pip                       23.1.2          py310h06a4308_0    defaults
pixman                    0.40.0               h7f8727e_1    defaults
pkginfo                   1.9.6           py310h06a4308_0    defaults
platformdirs              2.5.2           py310h06a4308_0    defaults
pluggy                    1.0.0           py310h06a4308_1    defaults
ply                       3.11            py310h06a4308_0    defaults
psutil                    5.9.0           py310h5eee18b_0    defaults
py-lief                   0.12.3          py310h6a678d5_0    defaults
pycosat                   0.6.4           py310h5eee18b_0    defaults
pycparser                 2.21               pyhd3eb1b0_0    defaults
pydotplus                 2.0.2                      py_3    defaults
pyjwt                     2.4.0           py310h06a4308_0    defaults
pyopenssl                 23.0.0          py310h06a4308_0    defaults
pyparsing                 3.0.9           py310h06a4308_0    defaults
pyqt                      5.15.7          py310h6a678d5_1    defaults
pyqt5-sip                 12.11.0                  pypi_0    pypi
pyrsistent                0.18.0          py310h7f8727e_0    defaults
pysocks                   1.7.1           py310h06a4308_0    defaults
python                    3.10.11              h7a1cb2a_2    defaults
python-dateutil           2.8.2              pyhd3eb1b0_0    defaults
python-fastjsonschema     2.16.2          py310h06a4308_0    defaults
python-libarchive-c       2.9                pyhd3eb1b0_1    defaults
pytz                      2022.7          py310h06a4308_0    defaults
pyyaml                    6.0             py310h5eee18b_1    defaults
qt-main                   5.15.2               h8373d8f_8    defaults
qt-webengine              5.15.9               hbbf29b9_6    defaults
qtpy                      2.2.0           py310h06a4308_0    defaults
qtwebkit                  5.212                h3fafdc1_5    defaults
readline                  8.2                  h5eee18b_0    defaults
requests                  2.29.0          py310h06a4308_0    defaults
requests-toolbelt         0.9.1              pyhd3eb1b0_0    defaults
ruamel.yaml               0.17.21         py310h5eee18b_0    defaults
ruamel.yaml.clib          0.2.6           py310h5eee18b_1    defaults
ruamel_yaml               0.17.21         py310h5eee18b_0    defaults
setuptools                67.8.0          py310h06a4308_0    defaults
sip                       6.6.2           py310h6a678d5_0    defaults
six                       1.16.0             pyhd3eb1b0_1    defaults
soupsieve                 2.4             py310h06a4308_0    defaults
sqlite                    3.41.2               h5eee18b_0    defaults
tk                        8.6.12               h1ccaba5_0    defaults
toml                      0.10.2             pyhd3eb1b0_0    defaults
tomli                     2.0.1           py310h06a4308_0    defaults
toolz                     0.12.0          py310h06a4308_0    defaults
tornado                   6.2             py310h5eee18b_0    defaults
tqdm                      4.65.0          py310h2f386ee_0    defaults
traitlets                 5.7.1           py310h06a4308_0    defaults
tzdata                    2023c                h04d1e81_0    defaults
ujson                     5.4.0           py310h6a678d5_0    defaults
urllib3                   1.26.16         py310h06a4308_0    defaults
wheel                     0.38.4          py310h06a4308_0    defaults
xz                        5.4.2                h5eee18b_0    defaults
yaml                      0.2.5                h7b6447c_0    defaults
zlib                      1.2.13               h5eee18b_0    defaults
zstandard                 0.19.0          py310h5eee18b_0    defaults
zstd                      1.5.5                hc292b87_0    defaults
```


### Additional Context

_No response_

**Comment by diegobersanetti** [comment]
Hello,

thanks for the update!

I confirm that the base environment only contains a basic selection of packages that i use.

I installed the ``conda-libmamba-solver``  as advised, then I tried the usual package update forcing one-time the use of such solver with ``--solver=libmamba``  parameter, and the issue was not found anymore.

The weird thing is switching back to the classic solver, either as default ot forcing it with `--solver=classic`, also made the issue disappear. it could be that the installation of the packages needed by the new solver somewhat fixed the corner case, but I cannot tell.
